### PR TITLE
Set up Go workflow and update testify

### PR DESCRIPTION
**Pull request by mmorel-35**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    - name: Build
+      run: go build -v ./...
+
+    - name: Test with coverage 
+      run: go test -race -v -count=1 -coverprofile=coverage.out ./...
+    - name: Upload Test Coverage
+      uses: codecov/codecov-action@v1
+      with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (mmorel-35) [inline .github/workflows/go.yml:47]: You'll need to provide the codecov_token in the project's secret

File: .github/dependabot.yml [hunk 2]
```diff
@@ -4,3 +4,8 @@ updates:
   directory: /
   schedule:
     interval: daily
+- package-ecosystem: github-actions
+  directory: /
+  schedule:
+    interval: daily
+
```
Comment (mmorel-35) [inline .github/dependabot.yml:11]:
```suggestion
```

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        continue-on-error: true
```
Comment (mmorel-35) [inline .github/workflows/go.yml:13]: This can be removed after linting warning has been fixed

File: .github/workflows/go.yml (hunk 1 above)
Comment (yurishkuro) [reply .github/workflows/go.yml:44]: added

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
Thanks, @mmorel-35 , the workflow ran successfully on master. Is there a way to suppress the deprecation warnings so that the lint can be set to fail the build?

**Comment by mmorel-35** [comment]
The warning can be ignored, but can't they be fixed instead ?

**Comment by yurishkuro** [comment]
The usage of the deprecated functions is in the tests and api_checker, they cannot be removed. #245

**Comment by mmorel-35** [comment]
Alright :)

**Pull request merged**